sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs

```diff
@@ -204,6 +209,7 @@ protected bool UseLocalDebugProxy
         protected RecordedTestBase(bool isAsync, RecordedTestMode? mode = null) : base(isAsync)
         {
             Mode = mode ?? TestEnvironment.GlobalTestMode;
+            AssetsJsonPath = GetAssetsJson();
```

Comment:
@JoshLove-msft I _believe_ this should be safe given that each test class should inherit from this guy.

It also means that we aren't running through a path traversal for every single recording start, which will actually save ourselves some disk flailing.